### DISCUSSION TITLE
Partial parsing of Tasmota SEN5x Data

### DISCUSSION THREAD

**Discussion by pmendiuk** [bellrichm/WeeWX-MQTTSubscribe#223] [Q&A]
I previously had SEN5x data from Tasmota12.5 going into WeeWX through MQTT Subscribe without issue.  We had a big wind storm and some of my ESP32 sensors did not survive the severe power glitches.  I flashed another ESP32 with the latest version of Tasmota(14.3) and now I am only getting part of the SEN5x data.  In my weeWX chart plots, there is no PM1, PM2.5, PM4 data.  There are differences in the TopicManager data log entries. 

Here is that topic captured by MQTT.fx:

tele/AQ1/SENSOR

{
  "Time" : 1732750923,
  "SEN5X" : {
    "PM1" : 16.1,
    "PM2.5" : 18.4,
    "PM4" : 19.7,
    "PM10" : 20.2,
    "NOx" : 1,
    "VOC" : 248,
    "Temperature" : 19.6,
    "Humidity" : 60.2,
    "DewPoint" : 11.6,
    "aHumidity" : 10.2735
  },
  "TempUnit" : "C"
}

Here are my weewx.conf entries for that topic:

                [[[tele/AQ1/SENSOR]]]
                        [[[[SEN5X PM1]]]]
                                name = aqi1PM1
                        [[[[SEN5X PM2.5]]]]
                                name = aqi1PM2_5
                        [[[[SEN5X PM4]]]]
                                name = aqi1PM4
                        [[[[SEN5X_PM10]]]]
                                name = aqi1PM10
                        [[[[SEN5X_NOx]]]]
                                name = aqi1NOX
                        [[[[SEN5X_VOC]]]]
                                name = aqi1VOC
                        [[[[SEN5X_Temperature]]]]
                                name = aqi1Temperature
                        [[[[SEN5X_Humidity]]]]
                                name = aqi1Humidity
                        [[[[SEN5X_DewPoint]]]]
                                name = aqi1DewPoint
                        [[[[SEN5X_aHumidity]]]]
                                name = aqi1AHumidity
                        [[[[TempUnit]]]]
                                ignore = True

Here are the journalctl log entries:

Nov 27 14:55:54 WeeWx1-debian12-vm python3[5392]: weewx[5392] DEBUG user.MQTTSubscribe: (Service) MessageCallbackProvider data-> incoming topic: tele/AQ1/SENSOR, QOS: 0, retain: 0, payload: b'{"Time":1732748153,"SEN5X":{"PM1":24.6,"PM2.5":28.3,"PM4":30.4,"PM10":31.3,"NOx":1,"VOC":298,"Temperature":19.8,"Humidity":59.3,"DewPoint":11.7,"aHumidity":10.2838},"TempUnit":"C"}'

Nov 27 14:55:54 WeeWx1-debian12-vm python3[5392]: weewx[5392] DEBUG user.MQTTSubscribe: (Service) TopicManager data-> incoming tele/AQ1/SENSOR: 'aqi1AHumidity': '10.2838', 'aqi1DewPoint': '11.7', 'aqi1Humidity': '59.3', 'aqi1NOX': '1.0', 'aqi1PM10': '31.3', 'aqi1Temperature': '19.8', 'aqi1VOC': '298.0', 'SEN5X_PM1': '24.6', 'SEN5X_PM2.5': '28.3', 'SEN5X_PM4': '30.4', 'Time': '1732748153.0'

**Comment by bellrichm** [comment]
It looks like you have a space (“ “) instead of an underscore (“_”) for these fields
```
            [[[tele/AQ1/SENSOR]]]
                    [[[[SEN5X PM1]]]]
                            name = aqi1PM1
                    [[[[SEN5X PM2.5]]]]
                            name = aqi1PM2_5
                    [[[[SEN5X PM4]]]]
                            name = aqi1PM4
```
In otherwords try ` [[[[SEN5X_PM1]]]]`, ` [[[[SEN5X_PM2.5]]]]`, and `[[[[SEN5X_PM4]]]]`

**Comment by pmendiuk** [comment]
Thanks for looking at it.  It wasn’t that clear in the ssh terminal window.
